src/main/scala/ciris/aws/ssm/package.scala

```diff
@@ -1,81 +1,50 @@
 package ciris.aws
 
+import cats.effect.Blocker
 import ciris._
-import ciris.api._
 import com.amazonaws.auth._
 import com.amazonaws.regions.Regions
-import com.amazonaws.services.simplesystemsmanagement.model.{
-  GetParameterRequest,
-  ParameterNotFoundException
-}
-import com.amazonaws.services.simplesystemsmanagement.AWSSimpleSystemsManagementClientBuilder
-
-import scala.util.{Try, Success}
+import com.amazonaws.services.simplesystemsmanagement._
+import com.amazonaws.services.simplesystemsmanagement.model._
 
 package object ssm {
+  final def param(key: String, blocker: Blocker)(
+    implicit region: Regions = Regions.EU_WEST_1,
+    credsProvider: AWSCredentialsProvider = new DefaultAWSCredentialsProviderChain()
+  ): ConfigValue[String] =
+    ConfigValue.blockOn(blocker) {
+      ConfigValue.suspend {
+        val ssmClient =
+          AWSSimpleSystemsManagementClientBuilder
+            .standard()
+            .withRegion(region)
+            .withCredentials(credsProvider)
+            .build()
+
+        val configKey =
+          ConfigKey(s"parameter $key from AWS SSM")
 
-  val SecretString: ConfigKeyType[String] =
-    ConfigKeyType[String]("secret string from AWS parameter store")
+        try {
+          val result =
+            ssmClient.getParameter {
+              new GetParameterRequest()
+                .withWithDecryption(true)
+                .withName(key)
+            }
 
-  def awsSsmSource(
-      region: Regions,
-      credsProvider: AWSCredentialsProvider
-  ): ConfigSource[Id, String, String] =
-    ConfigSource.fromTryOption(SecretString) { key =>
-      val ssmClient = AWSSimpleSystemsManagementClientBuilder
-        .standard()
-        .withRegion(region)
-        .withCredentials(credsProvider)
-        .build()
-      try {
-        val request = new GetParameterRequest()
-          .withName(key)
-          .withWithDecryption(true)
-        Try(ssmClient.getParameter(request))
-          .map { result =>
+          val loaded =
             for {
-              param <- Option(result.getParameter)
-              value <- Option(param.getValue)
-            } yield value
-          }
-          .recover {
-            case _: ParameterNotFoundException => None
-          }
-      } finally {
-        ssmClient.shutdown()
+              parameter <- Option(result.getParameter)
+              value <- Option(parameter.getValue)
+            } yield ConfigValue.loaded(configKey, value)
```

Comment:
Not entirely sure why these are wrapped in `Option`, and whether that means the parameter/value is missing, but this is consistent with previous behaviour.

Reply:
I think I was just being super-defensive because I was dealing with a Java API. Probably not necessary but let's leave it just in case.